src/test/java/spoon/test/ctType/CtTypeTest.java

```diff
@@ -267,4 +268,21 @@ public void testTypeDeclarationToReferenceRoundTripInNamedModule() {
 
 		assertSame(reFetchedTypeDecl, typeDecl);
 	}
+
+	@Test
+	public void testGetOrCreateCompilationUnit() {
+		// contract: the compilation unit should be returned using the shortcut method
+		final Launcher launcher = new Launcher();
+		final Factory factory = launcher.getFactory();
+
+		// create the elements required
+		CtCompilationUnit expectedCu = factory.createCompilationUnit();
+		CtClass<?> klass = factory.createClass("CompilationUnit");
```

Comment:
Not sure which is the best way here. I could either create a test file and add it to the launcher or just use factory methods. I think both will do a similar task but the latter was shorter to write.

Reply:
I tested for a `CtClass` instance in `CtType`. Is this alright? Or should I move this test case to `CtClassTest`? Moroever, do we also need to test for `CtInterface` because they are also `CtType` implementations?